### ISSUE TITLE
two questions about the function G

### ISSUE THREAD

**Issue by liujianzhao6328057** [GT-RIPL/L2C#8]
Hi, i have two questions about the function G:
1. What is the accuracy (the ability to predict the similarity between two images) of the function G ?
2.  The  "sampling in the Omniglot dataloader" makes sure that each cluster samples equal number of images. However, assuming that there are 5 classes per batch and each classes has 10 images,  there are 2500 pairs in total (500 similar pairs and 2000 dissimilar pairs). We can see that the imbalanced problem still exists.

**Comment by yenchanghsu** [comment]
1. The precision and recall of G are printed to console if you run this [script](https://github.com/GT-RIPL/L2C/blob/master/scripts/exp_unsupervised_transfer_Omniglot.sh#L2). The performance of G learned with Omniglot-bg and tested with Omniglot-eval or MNIST are available in Appendix Table 7 and 13 of the [paper](https://arxiv.org/pdf/1711.10125.pdf).
2. The sampler does not aim to solve the imbalance problem completely. The sampler only makes the imbalance problem less severe to make sure the number of similar pairs in a mini-batch is not zero. If the imbalance is a concern in your case, a simple option is to calibrate the threshold of binarization after the G is learned.

**Comment by linhlt-it-ee** [comment]
Can you please tell me how you converted [sij=0]P(Yi|xi,theta)PYj|xj,theta) to (1-sij)log(1-f(xi,theta)Tf(fxj,theta))
It seems like P(Yi|xi,theta)PYj|xj,theta) ->1-f(xi,theta)Tf(fxj,theta) for no reason.
It conflicts with the theory when 2 classes are differents. Your loss function has a positive class =1 when 2 samples have the same predictions. It means the higher overlap, the loss should be lower. In your case, the lower of overlap, the loss is lower.

**Comment by yenchanghsu** [comment]
Hi linhlt-it-ee, your question was answered in [another thread](https://github.com/GT-RIPL/L2C/issues/10).
